libraries/core_libs/network/include/network/tarcap/shared_states/syncing_state.hpp

```diff
@@ -62,9 +63,9 @@ class SyncingState {
   std::atomic<bool> deep_pbft_syncing_{false};
   std::atomic<bool> pbft_syncing_{false};
 
-  ExpirationCache<dev::p2p::NodeID> malicious_peers_;
+  std::unordered_map<dev::p2p::NodeID, std::chrono::steady_clock::time_point> malicious_peers_;
```

Comment:
now it is not thread-safe, mutex is needed

Reply:
map is now threadsafe